### ISSUE TITLE
Valgrind support for the Garbage Collector and JIT memory

### ISSUE THREAD

**Issue by mgaudet** [eclipse/omr#826]
_This issue is derived from [our Google Summer of Code 2017 Ideas List](https://wiki.eclipse.org/Google_Summer_of_Code_2017_Ideas#Eclipse_OMR)_ 
> Valgrind support for the Garbage Collector and JIT memory http://valgrind.org/docs/manual/mc-manual.html#mc-manual.mempools

**Comment by Varun-garg** [comment]
I am sharing my results for Valgrind on sample gctest to help get started with this issue. (ubuntu, gcc 5.4.0)

```
$ valgrind ./omrgctest -configListFile=fvtest/gctest/configuration/fvConfigListFile.txt
==38115== Memcheck, a memory error detector
==38115== Copyright (C) 2002-2015, and GNU GPL'd, by Julian Seward et al.
==38115== Using Valgrind-3.11.0 and LibVEX; rerun with -h for copyright info
==38115== Command: ./omrgctest -configListFile=fvtest/gctest/configuration/fvConfigListFile.txt
==38115== 
[==========] Running 8 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 8 tests from configFile/GCConfigTest
[ RUN      ] configFile/GCConfigTest.test/0
==38115== Warning: set address range perms: large range [0x64cd000, 0x264dc000) (noaccess)
==38115== Warning: set address range perms: large range [0x64cd000, 0x264dc000) (noaccess)
[       OK ] configFile/GCConfigTest.test/0 (3403 ms)
[ RUN      ] configFile/GCConfigTest.test/1
==38115== Warning: set address range perms: large range [0x64cd000, 0x264dc000) (noaccess)
==38115== Warning: set address range perms: large range [0x64cd000, 0x264dc000) (noaccess)
[       OK ] configFile/GCConfigTest.test/1 (3264 ms)
[ RUN      ] configFile/GCConfigTest.test/2
[       OK ] configFile/GCConfigTest.test/2 (1879 ms)
[ RUN      ] configFile/GCConfigTest.test/3
[       OK ] configFile/GCConfigTest.test/3 (2113 ms)
[ RUN      ] configFile/GCConfigTest.test/4
[       OK ] configFile/GCConfigTest.test/4 (1507 ms)
[ RUN      ] configFile/GCConfigTest.test/5
[       OK ] configFile/GCConfigTest.test/5 (2020 ms)
[ RUN      ] configFile/GCConfigTest.test/6
[       OK ] configFile/GCConfigTest.test/6 (1230 ms)
[ RUN      ] configFile/GCConfigTest.test/7
[       OK ] configFile/GCConfigTest.test/7 (1222 ms)
[----------] 8 tests from configFile/GCConfigTest (16650 ms total)

[----------] Global test environment tear-down
[==========] 8 tests from 1 test case ran. (16686 ms total)
[  PASSED  ] 8 tests.
[  ALL TESTS PASSED  ] 
==38115== 
==38115== HEAP SUMMARY:
==38115==     in use at exit: 168,050,208 bytes in 291,408 blocks
==38115==   total heap usage: 329,282 allocs, 37,874 frees, 212,062,117 bytes allocated
==38115== 
==38115== LEAK SUMMARY:
==38115==    definitely lost: 167,899,680 bytes in 291,386 blocks
==38115==    indirectly lost: 66,624 bytes in 5 blocks
==38115==      possibly lost: 11,200 bytes in 16 blocks
==38115==    still reachable: 72,704 bytes in 1 blocks
==38115==         suppressed: 0 bytes in 0 blocks
==38115== Rerun with --leak-check=full to see details of leaked memory
==38115== 
==38115== For counts of detected and suppressed errors, rerun with: -v
==38115== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

**Comment by Varun-garg** [comment]
Valgrind support for Garbage collector is merged in 876525fe341fff6f697b5c689f8a082e1356dee4
**PR:** #1331

**Comment by mgaudet** [comment]
Closed based in #1311
